lib/ansible/module_utils/network/asa/asa.py

```diff
@@ -81,22 +83,29 @@ def check_args(module):
 
 
 def get_connection(module):
-    global _CONNECTION
-    if _CONNECTION:
-        return _CONNECTION
-    _CONNECTION = Connection(module._socket_path)
+    if hasattr(module, '_asa_connection'):
+        return module._asa_connection
 
-    # Not all modules include the 'context' key.
-    context = module.params.get('context')
+    capabilities = get_capabilities(module)
+    network_api = capabilities.get('network_api')
+    if network_api == 'cliconf':
+        module._asa_connection = Connection(module._socket_path)
+    else:
+        module.fail_json(msg='Invalid connection type %s' % network_api)
 
-    if context:
```

Comment:
Removing the `context` related logic might break existing asa modules

Reply:
+1